zookeeper/core/component.py

```diff
@@ -271,6 +271,9 @@ def wrapped_fn(instance, name, value):
                             "Component instances can only be set as values if they are "
                             "not yet configured."
                         )
+                else:
+                    # If value is not a component, we can directly add this to in_scope.
+                    instance.__component_fields_with_values_in_scope__.add(name)
```

Comment:
What's the reason to have this behind an `else`? Does the code already work in the case of a component?

Reply:
The error was thrown [here](https://github.com/larq/zookeeper/blob/master/zookeeper/core/component.py#L691), which is only run for non-component `Fields`. However, now that you mention it it seems we had a missing `configure()` in the test that tries this with a `ComponentField`, and it throws a lovely assertion error. I'll investigate.

Reply:
Alright, this indeed needed to be done for `ComponentField`s too, because it would otherwise revert to a default instance of that class. The modified test now passes again :+1: